test/metabase/parameters/params_test.clj

```diff
@@ -118,7 +119,66 @@
                                           :dimensions         []}]}
               (-> (t2/hydrate dashboard :param_fields)
                   :param_fields
-                  mt/derecordize))))))
+                  mt/derecordize)))))
+  (testing "should ignore parameter targets with incorrect :stage-number"
+    (mt/with-temporary-setting-values [enable-public-sharing true]
+      (mt/with-temp [:model/Dashboard    dashboard  {:parameters [{:id "p1" :type :number/=}
+                                                                  {:id "p2" :type :number/=}
+                                                                  {:id "p3" :type :number/=}
+                                                                  {:id "p4" :type :number/=}
+                                                                  {:id "p5" :type :string/=}
+                                                                  {:id "p6" :type :string/=}
+                                                                  {:id "p7" :type :number/=}]}
+                     :model/Card          card      {:dataset_query
+                                                     {:database (mt/id)
+                                                      :type     :query
+                                                      :query    {:source-table (mt/id :products)
+                                                                 :aggregation [[:count]]
+                                                                 :breakout [[:field (mt/id :products :category) nil]]}}}
```

Comment:
This is better than with `mt/mbql-query`, IMO.

Reply:
Fixed